src/main/java/com/iexec/resultproxy/proxy/ProxyService.java

```diff
@@ -154,14 +155,7 @@ boolean isResultFound(String chainTaskId) {
         return ipfsResultService.doesResultExist(chainTaskId);
     }
 
-    String addResult(Result result, byte[] data) {
-        if (result == null || result.getChainTaskId() == null) {
-            return "";
-        }
-        return ipfsResultService.addResult(result, data);
-    }
-
-    Optional<byte[]> getResult(String chainTaskId) {
-        return ipfsResultService.getResult(chainTaskId);
+    String addResult(ResultModel model) {
+        return ipfsResultService.addResult(model.getChainTaskId(), model.getZip());
```

Comment:
Do we assume `model` is never null?

Reply:
`model` is the request body. I will need to check if it can be null.
In case of null check, I would prefer to implement it at the controller level.